compiler/src/dotty/tools/dotc/transform/Splicing.scala

```diff
@@ -77,7 +77,7 @@ class Splicing extends MacroTransform:
 
   override def run(using Context): Unit =
     if ctx.compilationUnit.needsStaging then
-      super.run(using freshStagingContext)
+      super.run
```

Comment:
Why are some of the calls have `freshStagingContext` removed, while some are replaced with `StagingLevel.freshContext`? I don't really understand why or when which is used.

Reply:
Oh it seems like `StagingLevel.freshContext` are used in the tests...?

Reply:
I changed some instances in `tests/pos-with-compiler-cc/dotc` by mistake. I reverted those changes. That test should not change.

Reply:
Note that I do not initialize the levels until we add the first symbol with `level != 0`.